investors/views.py

```
@@ -215,6 +218,156 @@ def perform_destroy(self, instance):
+                )
+            detail = getattr(e, "detail", {})
+            if isinstance(detail, dict):
+                for field, msgs in detail.items():
+                    if isinstance(msgs, list):
+                        msgs = [msgs]
+                    if any("own project" in str(m).lower() for m in msgs):
+                        logger.warning(
+                            "FollowedProject create failed: own project",
+                            extra={"project_id": project_id, "by_user": getattr(user, "pk", None)},
+                        )
+            raise
+
+        self.perform_create(serializer)
+        headers = self.get_success_headers(serializer.data)
+        return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
+
+    def perform_create(self, serializer):
+        user = self.request.user
+        project = serializer.validated_data.get("project")
+        if project is None:
+            logger.warning("FollowedProject create failed: missing project", extra={"by_user": user.pk})
+            raise ValidationError({"project": "This field is required."})
+
+        startup_user_id = getattr(getattr(project, "startup", None), "user_id", None)
+        if startup_user_id == user.pk:
+            logger.warning(
+                "FollowedProject create failed: own project",
+                extra={"project_id": getattr(project, "pk", None), "startup_user_id": startup_user_id, "by_user": user.pk},
+            )
+            raise ValidationError({"project": "You cannot follow your own project."})
+
+        try:
+            instance = serializer.save(investor=user.investor)
+        except IntegrityError:
+            logger.warning(
+                "FollowedProject create failed: duplicate",
+                extra={"investor_id": user.investor.pk, "project_id": project.pk, "by_user": user.pk},
+            )
+            raise ValidationError({"non_field_errors": ["Already followed."]})
```

The exception handler for IntegrityError does not specify the exception to catch, which may mask other errors. Import and catch IntegrityError explicitly.
```
from django.db import IntegrityError

try:
    instance = serializer.save(investor=user.investor)
except IntegrityError:
    # handle duplicate
```